src/main/java/io/github/vhoyon/bot/utilities/music/AudioListener.java

```diff
@@ -70,26 +79,46 @@ public boolean nextTrack(){
 	@Override
 	public void onTrackEnd(AudioPlayer player, AudioTrack track,
 			AudioTrackEndReason endReason){
-
-			this.player.getCommand().remember(track.getInfo().uri, "LATEST_SONG");
-
-		if(this.player.getCommand().hasMemory("LOOP_ONCE") && (boolean) this.player.getCommand().getMemory("LOOP_ONCE") ){
+		
+		this.player.getCommand().remember(track.getInfo().uri, "LATEST_SONG");
+		
+		if(this.player.getCommand().hasMemory("LOOP_ONCE")
+				&& (boolean)this.player.getCommand().getMemory("LOOP_ONCE")){
 			player.playTrack(track.makeClone());
-		}else if(endReason.mayStartNext){
-			if((this.player.getCommand().hasMemory("MUSIC_LOOP") && (boolean) this.player.getCommand().getMemory("MUSIC_LOOP"))) {
+		}
+		else if(endReason.mayStartNext){
+			if((this.player.getCommand().hasMemory("MUSIC_LOOP") && (boolean)this.player
+					.getCommand().getMemory("MUSIC_LOOP"))){
 				tracks.add(track.makeClone());
 				nextTrack();
-			}else{
-				if (!tracks.isEmpty()) {
+			}
+			else{
+				if(!tracks.isEmpty()){
 					nextTrack();
 				}
+				else{
+					
+					Runnable task = new Runnable() {
+						@Override
+						public void run() {
+							MusicManager.get().emptyPlayer(AudioListener.this.player.getCommand());
+							AudioListener.this.player.getCommand().sendInfoMessage(
+								"Disconnected due to inactivity");
+						}
+					};
+					TimerManager.schedule("noMusicDelay",5000, task);
```

Comment:
Don't forget the make this delay tweakable by settings (view the CommandSetting to further understand how to do it).

As described in #114, the default would be `30000` milliseconds.